jenkins/release_ci/opensearch_maven_release/maven-sign-release.jenkinsfile

```diff
@@ -0,0 +1,94 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(
+            defaultValue: 'x64',
+            name: 'ARCH',
+            trim: true
+        )
+        string(
+            defaultValue: 'linux',
+            name: 'PLATFORM',
+            trim: true
+        )
+        string(
+            name: 'BUILD_ID',
+            trim: true
+        )
+        string(
+            name: 'VERSION',
+            trim: true
+        )
+    }
+    environment {
+        ARTIFACT_PATH = "distribution-build-opensearch/${params.VERSION}/${params.BUILD_ID}/${params.PLATFORM}/${params.ARCH}/builds"
+    }
+    stages {
+        stage('sign') {
+            steps {
+                script {
+                    sh 'echo "Downloading from S3."'
+                    downloadFromS3(
+                        sourcePath: "$WORKSPACE/artifacts",
+                        bucket: "${ARTIFACT_BUCKET_NAME}",
+                        path: "${ARTIFACT_PATH}/",
+                        force: true
+                    )
+                    sh 'echo "Signing Maven artifacts."'
+                    signatureType = '.asc'
+                    signArtifacts(
+                        artifactPath: "$WORKSPACE/artifacts/$ARTIFACT_PATH/opensearch/manifest.yml",
+                        signatureType: signatureType,
+                        component: '',
+                        type: 'maven'
+                    )
+                    uploadToS3(
+                        sourcePath: "$WORKSPACE/artifacts/${ARTIFACT_PATH}/opensearch/maven",
+                        bucket: "${ARTIFACT_BUCKET_NAME}",
+                        path: "${ARTIFACT_PATH}/maven-signed"
+                    )
+                }
+            }
+        }
+        stage('stage maven artifacts') {
+            tools {
+                maven "maven-3.8.2"
+            }
+            environment {
+                REPO_URL = "https://aws.oss.sonatype.org/"
+                STAGING_PROFILE_ID = "${SONATYPE_STAGING_PROFILE_ID}"
+                BUILD_ID = "${params.BUILD_ID}"
+            }
+            steps {
+                script {
+                    git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+                    downloadFromS3(
```

Comment:
Do we need to upload the Artifacts on s3? Since we are downloading them again here.

Reply:
That makes sense. In the same workspace, both upload and this download are not needed.